### PR TITLE
Adds support for Node to use Interface's DataModel instead of relying on Aliases

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

For a while, we have had to create `alias` references to DataSources for models that want to use `node` but are implementations of a root `interface`. With this PR, the `node` resolver will be smart enough to "work it's way up the hierarchy chain" and resolve without needing to alias every implementation. 

This PR is also smart enough that specific instances of an implementation that supply their own datasource will override default behavior. 

I also improved the error message for instances where a dataSource is present, but the dataSource does not implement a getFromId function.

### What design trade-offs/decisions were made?

n/a

### How do I test this PR?

Run the app, you shouldn't see any changes.

### What automated tests did you write?

N/a. Fixed some existing tests that were pulling bad data prior to this PR.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [x] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,4 +120,4 @@ static propTypes = {
 };
 ```
 
-🍕🍕🍕
\ No newline at end of file
+🍕🍕🍕🍕
```
Comment (redreceipt) [inline README.md]: @vinnyjth please explain this

Comment (vinnyjth) [reply README.md]: I need to trigger another build on travis and the "rebuild" button wasn't working.

Comment (redreceipt) [reply README.md]: ha! oh

File: packages/apollos-server-core/src/node/__tests__/__snapshots__/model.js.snap [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Node Node class should throw error if it can't find a matching model 1`] = `"No dataSource found using NoModel"`;
```
Comment (IsaacHardy) [inline packages/apollos-server-core/src/node/__tests__/__snapshots__/model.js.snap]: Is this the expected outcome?

File: packages/apollos-server-core/src/node/model.js [hunk 3]
```diff
@@ -29,19 +30,36 @@ export function parseGlobalId(encodedId) {
   }
 }
 
+const getPossibleDataModels = ({ schema, __type }) => {
```
Comment (IsaacHardy) [inline packages/apollos-server-core/src/node/model.js]: This whole method is hard to comprehend. Inline comments might be useful to help explain some of the work happenings.

File: packages/apollos-server-core/src/node/__tests__/__snapshots__/model.js.snap (hunk 2 above)
Comment (vinnyjth) [reply packages/apollos-server-core/src/node/__tests__/__snapshots__/model.js.snap]: Yes

File: packages/apollos-server-core/src/node/model.js (hunk 3 above)
Comment (vinnyjth) [reply packages/apollos-server-core/src/node/model.js]: Can you expand on what you find hard to comprehend about it? I don't doubt you for a second, but I want to make sure that I leave the right comments. I tried to name the variables in such a way that the code was self documenting.

Comment (IsaacHardy) [reply packages/apollos-server-core/src/node/model.js]: Line 34 seems to be grabbing the possibleType of a node, but I have no idea what `.getTypeMap()` is doing.

Line 35 has some very interesting properties to check: `possibleType.astNode.interfaces`

Line 38 as well ^

Line 42 has an array with possibleInterfaces, and I have no idea what that even looks like.

Comment (vinnyjth) [reply packages/apollos-server-core/src/node/model.js]: 👍 adds more helpful comments

**Pull request merged**